projectile.el

```diff
@@ -3429,7 +3442,9 @@ running the command."
          (command (projectile-maybe-read-command show-prompt
                                                  command
                                                  prompt-prefix)))
-    (puthash default-directory command command-map)
+    (when command-map
+      (puthash default-directory command command-map)
+      (ring-insert (projectile--get-command-history default-directory) command))
```

Comment:
Why is this not `projectile-project-root' rather than `default-directory'

Reply:
This is the right idea but needs to catch some errors -- if there is not last, it crashes.

```
Debugger entered--Lisp error: (wrong-type-argument stringp nil)
  string-match("\\`\\s *cd\\(?:\\s +\\(\\S +?\\|'[^']*'\\|\"\\(?:[^\"`$\\]\\|\\\\.\\)*\"\\)\\)?\\s *[;&\n]" nil)
  compilation-start(nil nil)
  compile(nil)
  (if (functionp cmd) (funcall cmd) (compile cmd))
  projectile-run-compilation(nil)
```

Also, it's seems to work if I am in a *compilation* buffer, but not elsewhere in the project. Would it not make more sense to hash by the project root than default directory?